src/hooks.js

```diff
@@ -0,0 +1,87 @@
+import {useEffect, useLayoutEffect} from 'react'
+import debounce from 'lodash.debounce'
+import countryIso from 'country-iso'
+import countryData from 'country-data'
+
+import drawChart from './chart'
+import getCountryData from './consumer'
+import geolocationStates from './geolocationStates'
+
+const google = window.google
+
+const getCountrySlug = coords => {
+  const code = countryIso.get(coords.latitude, coords.longitude)
+  return countryData.countries[code].name
+}
+
+const setUpGeolocation = async ({coords, setGeolocationState, setSelectedCountries, setData}) => {
+  const slug = getCountrySlug(coords).toLowerCase()
+  setGeolocationState(geolocationStates.loading)
+  google.charts.setOnLoadCallback(async () => {
+    const localData = await getCountryData(slug)
+    setGeolocationState(geolocationStates.loaded)
+    setSelectedCountries([slug])
+    setData({ [slug]: localData })
+  })
+}
+
+const handleGeoError = async ({code, fallbackCountry, setGeolocationState, setSelectedCountries, setData}) => {
```

Comment:
If the Geolocation access is denied by the browser on page load, the application crashes with:
```
TypeError: google.visualization.arrayToDataTable is not a function
```

Reply:
I don't seem to be able to replicate that.

Reply:
Let me try again

Reply:
I can still reproduce it like this:
- Start the dev server
- Set the Geolocation access to "Ask"
- Reload the page
- When it asks for Geolocation access, hit "Allow"
- Set Geolocation access to "Deny" manually
- Reload the page

Reply:
Interestingly I get a different error (`TypeError: google.visualization.LineChart is not a constructor`) with these steps:
- Start the dev server
- Set the Geolocation access to "Ask"
- Reload the page
- When it asks for Geolocation access, hit "Block"
- Reload the page

Reply:
It looks like `drawChart` is called before `google.charts.setOnLoadCallback` is triggered.

Maybe it would make sense to create a totally separate hook for the loading state of the Google charts which has only a `useState` initialized to false and set to true on `google.charts.setOnLoadCallback`.

Reply:
Turns out the problem was that I wasn't calling `google.charts.setOnLoadCallback` at call in the `handleGeoError` function.

Should be fixed. Please verify and resolve.